stdlib/memory.wat

```diff
@@ -1,22 +1,147 @@
 (module
   (memory (import "js" "mem") 1)
+  (func $print_str (import "imports" "print_str") (param i32) (result i32))
   (global $heap (mut i32) (i32.const 4))
 
+
   ;; Take an amount of blocks (4-byte words) to allocate, return an address
   ;; handle suitable for giving to other access methods
-  (func (export "alloc") (param $amount i32) (result i32)
+  (func $alloc (export "alloc") (param $amount i32) (result i32)
     (local $addr i32)
     (local.set $addr (global.get $heap))
     (global.set $heap (i32.add (global.get $heap) (i32.mul (local.get $amount) (i32.const 4))))
     (local.get $addr))
 
+  (func $duplicate_str(export "duplicate_str") (param $source i32) (param $dest i32)
```

Comment:
I think these should be in a different module than `memory.wat` – that should _just_ be for the memory operations. Perhaps a new file called `string.wat`?

Reply:
Sure.